src/App.js

```diff
@@ -1,57 +1,63 @@
 import React, { Component } from 'react';
-import { Formik } from 'formik';
 import { Form, Icon, Input, Button } from 'antd';
-import './App.css';
 
-class SubscriptionForm extends React.Component {
-  constructor(props) {
-    super(props);
-  }
-
-  getFieldDecorator = e => {
+function hasErrors(fieldsError) {
+  return Object.keys(fieldsError).some(field => fieldsError[field]);
+}
 
+class HorizontalLoginForm extends React.Component {
+  componentDidMount() {
+    // To disable submit button at the beginning.
+    this.props.form.validateFields();
   }
 
-  getFieldsError = e => {
-
-  }
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.form.validateFields((err, values) => {
+      if (!err) {
+        console.log('Received values of form: ', values);
+      }
+    });
+  };
 
   render() {
-    const usernameError = '';
-    const passwordError = '';
-    const hasErrors = false;
+    const { getFieldDecorator, getFieldsError, getFieldError, isFieldTouched } = this.props.form;
 
+    // Only show error after a field is touched.
+    const usernameError = isFieldTouched('username') && getFieldError('username');
+    const passwordError = isFieldTouched('password') && getFieldError('password');
     return (
-      <Form onSubmit={this.handleSubmit}>
+      <Form onSubmit={this.handleSubmit} className="form-subscribe">
```

Comment:
Long form at one place. hard to follow. 
Move out into separated component

Reply:
"Validation decorators" should be separated as well. at least move them to variables